src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -983,4 +983,24 @@ public static boolean isValidHostAuthority(String host) {
     int len = host.length();
     return HostAndPortImpl.parseHost(host, 0, len) == len;
   }
+
+  public static boolean canUpgradeToWebSocket(HttpServerRequest req) {
+    if (req.version() != io.vertx.core.http.HttpVersion.HTTP_1_1) {
+      return false;
+    }
+    if (req.method() != io.vertx.core.http.HttpMethod.GET) {
+      return false;
+    }
+    MultiMap headers = req.headers();
+    for (String connection : headers.getAll(io.vertx.core.http.HttpHeaders.CONNECTION)) {
+      if (connection.toLowerCase().contains(io.vertx.core.http.HttpHeaders.UPGRADE)) {
+        for (String upgrade : headers.getAll(io.vertx.core.http.HttpHeaders.UPGRADE)) {
```

Comment:
shouldn't we check for the exact `Upgrade: websocket`? it's cheaper and we can always address in future special cases.

Reply:
No, because servers should accept any case according to the spec